src/ussd_optout_rapidpro.js

```diff
@@ -1,29 +1,333 @@
+                choices: [
+                    new Choice("yes", $("Next"))
+                ],
+                next: "state_trigger_rapidpro_flow"
+            });
+        });
+
+        self.states.add("state_check_previous_optout", function(name) {
+            // Skip this state if they haven't opted out
+            //Skip this state if they have an opted_out reason
+            var contact = self.im.user.get_answer("contact");
+            var optout_reason = contact.fields.optout_reason;
+            if (!self.contact_in_group(self.im.user.get_answer("contact"), self.im.config.optout_group_ids)) {
+                return self.states.create("state_no_previous_optout");
+            }
+            
+            else if((self.contact_in_group(self.im.user.get_answer("contact"), self.im.config.optout_group_ids)) && optout_reason === "") {
+                var question = "Welcome MomConnect. You've opted out of receiving messages from us. Please tell us why:";
+                return new PaginatedChoiceState(name, {
+                    question: $(question),
+                    error: $(question),
+                    accept_labels: true,
+                    options_per_page: null,
+                    characters_per_page: 160,
+                    choices: [
+                        new Choice("miscarriage", $("Miscarriage")),
+                        new Choice("stillborn", $("Baby was stillborn")),
+                        new Choice("babyloss", $("Baby passed away")),
+                        new Choice("no", $("Msgs aren't helpful")),
+                        new Choice("no", $("Other")),
+                        new Choice("no", $("I prefer not to say"))
+                    ],
+                    next: function(choice) {
+                        self.im.user.set_answer("optout_reason", choice.value);
+                        if(_.includes(["miscarriage", "stillborn", "babyloss"], choice.value)) {
+                            return "state_loss_optout";
+                        } else {
+                            return "state_nonloss_optout";
+                        }  
+                    }  
+                });
+            }
+        });
+
+        self.add("state_trigger_rapidpro_flow", function(name, opts) {
+            var msisdn = utils.normalize_msisdn(self.im.user.addr, "ZA");
+            return self.rapidpro.start_flow(self.im.config.flow_uuid, null, "tel:" + msisdn, {
+                babyloss_subscription: 
+                    self.im.user.get_answer("state_loss_optout") === "yes" ? "TRUE" : "FALSE",
+                delete_info_for_babyloss: 
+                    self.im.user.get_answer("state_delete_info_for_loss") === "yes" ? "TRUE" : "FALSE",
+                delete_info_consent: 
+                        self.im.user.get_answer("state_delete_research_info") === "yes" ? "TRUE" : "FALSE",
+                optout_reason: self.im.user.get_answer("optout_reason"),
+                language: self.im.user.lang,
```

Comment:
The user isn't changing there language in this flow, so we don't need to send the user's language to RapidPro, RapidPro already knows what the user's language is

Reply:
will remove it